Properly skip `pod install` on non-Mac platforms

bryanstearns opened the pull request:
... by running a script (`maybe-pod-install`) that silently succeeds if `pod` can't be found.

I've tested `ignite new MyApp -b ./bowser` on Mac and Windows, and with `yarn` installed on Windows, it succeeded. Without `yarn` installed, `npm i` couldn't find the new script - I don't know why. I didn't test further than generating a new app on Windows.

Note that I also didn't test without Yarn on Mac - this might've broken the local-boilerplate for non-yarn users (and I'm loathe to screw up my Mac development environment). Any hard-no-on-Yarn users want to test this for me?

This should fix #243 and maybe #244.

jamonholmgren commented on package.json:
```diff
@@ -15,6 +15,9 @@
     "vanilla"
   ],
+  "bin": {
+    "maybe-pod-install": "bin/maybe-pod-install"
```
Do we need to do `./bin/maybe-pod-install` to be slightly safer?

bryanstearns replied:
Nope, here's why: The `maybe-pod-install` script is part of the `ignite-bowser` package; it's not installed in the target app at all, but since the target app has the boilerplate package as a dependency, (and here's where I'm handwaving a little because I don't understand the mechanism) node takes care of making sure executables provided by dependency packages are in the path. Somehow.

 At the point where yarn is running (whether now, or later whenever the developer runs it), the current directory is the target app, so I'm depending on that mechanism to locate `maybe-pod-install`, just like it locates `solidarity` and `jetify` already.

(I'm assuming something's wrong with this mechanism in `npm` when the dependency is file-system-relative, as it is when we use the `-b` option with a path to the boilerplate, but `yarn` seems to get it right, and since we strongly recommend `yarn`, I'm not going to dig further now.)

wistler commented:
@bryanstearns Yes, this fixes #243

bryanstearns commented:
@wistler Thanks for testing this!

bryanstearns commented:
I had another idea about this problem: tonight I added another command to a project's postInstall, to kill any running instance of the Metro bundler with `pkill -f "cli.js start"`, since I always want to restart Metro if I've changed packages (or projects!). 

This won't work on Windows, of course, but even if I did, I'd probably end up adding another executable to ignite-bower's `bin` directory as I did with this PR.

Instead, would it be better to add a single javascript script to the new application itself that would do all the postinstall steps? It'd be easier to do platform-specific things there. (This would also avoid the npm problem I mentioned in my reply above.)

bryanstearns commented:
Closing this in favor of #247, a better strategy.

zucchinho commented:
> Closing this in favor of #247, a better strategy.

it is a better strategy, but this approach is perfectly acceptable to patch the current release, which is broken on non-mac os. would strongly encourage applying this PR as a patch, while you iron out the kinks in the other approach (no activity on #247 for 11 days atow). going by many responses on the slack community, this is deterring developers who are just getting acquainted with ignite-bowser, since it falls at the first hurdle. personally, i just checked out this commit and pointed ignite at my clone. a minor inconvenience, admittedly, but will absolutely put developers off using the bp, when they otherwise may have got a lot of value out of it.
